### ISSUE TITLE
Mac OS testing and builds

### ISSUE THREAD

**Issue by marioortizmanero** [vidify/vidify#98]
Mac OS support currently is just theoretical... I haven't tried it since v1.0, so I should get an emulator or something like that. Automating the builds would help a lot, too.

**Comment by sammyviv** [comment]
Hi, Vidify is truly a unique idea and impressive piece of code!! I did get it working on Mac and loving it so far. But it's just in the terminal right now with a python window for the video, which does leave much to be desired.

I would help in any way I can for testing and support if we can get it to a full fledged App like the Windows and Linux versions.

It works pretty well even now, so core code changed are not required, just getting a UI running and packaged in a App Bundle would do wonders.

**Comment by marioortizmanero** [comment]
I'm glad you like the project! I haven't been able to work on it much this month as I've been quite busy with finals, but I'll retake it once the semester is over. I agree with you, I still have to work on the GUI. I'm not a big fan of programming them but it's necessary so that Vidify reaches out to users other than programmers. I should specially work on the quick menu and the subtitles so that the terminal isn't even necessary.

I actually do have a Mac around, so I'll test it myself soon and integrate the CI setup I have for Linux and Windows right now. As far as testing goes, trying out every feature of the app should be enough. As in, making sure every music player and video player work.

I also wanted to upload Vidify to snapcraft on Linux (but it's been a pain). By App Bundle you mean the typical installer for Mac OS, right? I'm not too sure about how that works, so I'll have to investigate a bit. Should I also upload it to Homebrew, or is that only for CLI apps and libraries?

**Comment by sammyviv** [comment]
Yeah, GUI coding is quite tedious so I don't blame you haha. Right now it seems you're using PyQt, which is quite tedious for making GUIs. Maybe you can look into Kivy, it's quite flexible and cross-platform support is much better. An idea for the GUI could be an integrated vertical window with the video playing up top and the lyrics displayed bellow in a dropdown tab.

I tested pretty much every feature on the Mac and it works perfectly. I had issues with the PyQt installation, it's quite picky with the versions but probably not a big issue.

Snapcraft is notorious for initial distributions, maybe try Flatpak?. For Mac OS, I don't think an installer is required, by App Bundle I meant just a regular .App which contains all resources and compiled binaries, py2app will handle all of it with little setup, it's quite robust and also has some GUI capabilities so it shouldn't be too much of a pain to get the App and GUI done together.

Homebrew can be used for GUI apps as well, but it's primarily for CLI based tools. Honestly homebrew is great for developers but regular users wouldn't even go near it so I don't feel it's worth the effort.

**Comment by marioortizmanero** [comment]
I don't think changing the GUI framework right now is a good idea. There's already a good amount of code already written, and it would take a while to rewrite it completely. Besides, I still prefer Qt because it's more popular and has better support overall (although I haven't tried Kivy).

My idea regarding the GUI was something similar to what can be found on #19 and this [project](https://github.com/vidify/vidify/projects/4). The quick menu can be opened to the left, and I suppose the lyrics could fit well to the right. The thing is that I don't know how that can be done with Qt because it's a floating window and those are a bit messy to work with.

Honestly flatpak might be the best choice here... Thanks for reminding me. The thing is that I'm not sure if it's as used as snap. And about the .App, I'm currently using PyInstaller (I have a bunch of scripts I already use at [vidify/dev](https://github.com/vidify/vidify/tree/master/dev)), which is kinda the successor of py2exe (now deprecated). It seems to work for Mac OS, too, so I'll first try using it, and try py2app if I fail. Thanks for the heads-up. It might not be worth it to upload it to Homebrew for now.

**Comment by sammyviv** [comment]
Yeah you're right about the framework, if you're comfortable with Qt then it doesn't make sense to change it at this point.

The idea you have for the GUI is pretty good... but the sidebar and options can be cleaned up alittle more. I'll try crafting a mockup which is more inline with the modern UI designs of Win10 and Mac OS, while keeping it modular.

No worries at all. I really appreciate your attention on this project cause it does have potential. Please do let me know if you need any help.

**Comment by marioortizmanero** [comment]
Great! Thanks a bunch for the help, I would love more feedback on this app; specially on the user-oriented part that's often forgotten in these kind of projects, and as I'm the only developer I don't get many opinions on it. I was working on an official [website](http://vidify.org), with the source files [hosted here](https://github.com/vidify/vidify.org-source) and the resulting site [here](https://github.com/vidify/vidify.org), to try and introduce it to anyone in a more user-friendly way. Let me know if there's anything you would change in there :)
